### PR TITLE
Helper to Get Provider from Top Frame

### PR DESCRIPTION
**Pull request by fan-zhang-sv**

### CODE REVIEW

File: packages/wallet-sdk/src/util/provider.ts [hunk 1]
```diff
@@ -43,6 +43,16 @@ export function getCoinbaseInjectedSigner(): Signer | undefined {
   return window.coinbaseWalletSigner;
 }
 
+function getTopFrameInjectedProvider(): CBInjectedProvider | undefined {
```
Comment (bangtoven) [inline packages/wallet-sdk/src/util/provider.ts:46]: how about calling it getInjectedProvider or something 
and encapsulate all window.ethereum logic?

**Review by bangtoven — COMMENTED** [review]
@fan-zhang-sv

File: packages/wallet-sdk/src/util/provider.ts (hunk 1 above)
Comment (fan-zhang-sv) [reply packages/wallet-sdk/src/util/provider.ts]: pushed, how does `getCipherWebViewInjectedProvider` sound?

Comment (bangtoven) [reply packages/wallet-sdk/src/util/provider.ts]: to be it seems to be just simple helper to retrieve injected provider from any of the window properties, either window or window.top
and cipher web view check happens the next line

Comment (fan-zhang-sv) [reply packages/wallet-sdk/src/util/provider.ts]: agree, `getInjectedEthereum`?

**Review by bangtoven — APPROVED** [review]

File: packages/wallet-sdk/src/util/provider.ts [hunk 2]
```diff
@@ -43,6 +43,16 @@ export function getCoinbaseInjectedSigner(): Signer | undefined {
...
 }
 
+function getInjectedEthereum(): CBInjectedProvider | undefined {
+  const window = globalThis as CBWindow;
+  try {
+    return window.ethereum ?? window.top?.ethereum;
+    // eslint-disable-next-line no-empty
+  } catch {}
+
+  return undefined;
```
Comment (bangtoven) [inline packages/wallet-sdk/src/util/provider.ts:53]: how about this?
```suggestion
  try {
    const window = globalThis as CBWindow;
    return window.ethereum ?? window.top?.ethereum;
  } catch {
    return undefined;
  }
```

File: packages/wallet-sdk/src/util/provider.ts [hunk 3]
```diff
@@ -43,6 +43,16 @@ export function getCoinbaseInjectedSigner(): Signer | undefined {
...
+}
+
 export function getCoinbaseInjectedProvider({
   metadata,
   preference,
```
Comment (bangtoven) [inline packages/wallet-sdk/src/util/provider.ts:58]: we might remove the line 50, 
and do this instead for line 56, `const extension =(globalThis as CBWindow).coinbaseWalletExtension;`

Comment (bangtoven) [reply packages/wallet-sdk/src/util/provider.ts:58]: or, seems bit overkill, but for the sake of consistency, we might create a dedicated helper method for this maybe?

**Pull request merged**